ion-c-sys/src/reader.rs

```diff
@@ -533,6 +552,44 @@ impl<'a> IonCReader for IonCReaderHandle<'a> {
             Ok(buf)
         }
     }
+
+    // *NOT* annotated with `#[position_error]` - if reading the current
+    // position is an error, then we don't want to include the position in the
+    // error!
+    //
+    // This method is actually implemented in `reader_current_pos`. Otherwise,
+    // various reader trait methods require additional borrows.
+    #[inline]
+    fn pos(&self) -> IonCResult<Position> {
+        reader_current_pos(&self.reader)
+    }
+}
+
+#[inline]
+fn reader_current_pos(reader: &hREADER) -> IonCResult<Position> {
+    let mut bytes: i64 = -1;
+    let mut line = -1;
+    let mut offset = -1;
+    ionc!(ion_reader_get_position(
+        *reader,
+        &mut bytes,
+        &mut line,
+        &mut offset
+    ))?;
+    Ok((bytes, line, offset).into())
+}
```

Comment:
Should this just be a method with the reader or does that complicate things?

Reply:
It compiles things because of the `move` closure generated by the macro. Want a comment?

Reply:
Might be good to comment just so someone doesn't stub their toe on this in the future.